front_end/models/react_native/ReactDevToolsBindingsModel.ts

```diff
@@ -54,10 +56,44 @@ export class ReactDevToolsBindingsModel extends SDK.SDKModel.SDKModel {
     }
 
     if (parsedMessage) {
-      this.dispatchMessageToDomainEventListeners(parsedMessage.domain, parsedMessage.message);
+      const domainName = parsedMessage.domain;
+
+      // Messages are flushed out when execution context is created and dispatcher is initialized
+      if (this.fuseboxDispatcherIsInitialized && this.isDomainMessagesQueueEmpty(domainName)) {
+        this.dispatchMessageToDomainEventListeners(domainName, parsedMessage.message);
+      } else {
+        // This could happen when backend is already sending messages via binding
+        // But ReactDevToolsBindingsModel is busy executing async tasks
+        this.queueMessageForDomain(domainName, parsedMessage.message);
+      }
     }
   }
 
+  private queueMessageForDomain(domainName: DomainName, message: JSONValue): void {
+    let queue = this.domainToMessageQueue.get(domainName);
+    if (!queue) {
+      queue = new Set();
+      this.domainToMessageQueue.set(domainName, queue);
+    }
+
+    queue.add(message);
+  }
+
+  private flushOutDomainMessagesQueues(): void {
+    for (const [domainName, queue] of this.domainToMessageQueue.entries()) {
+      for (const message of queue) {
+        this.dispatchMessageToDomainEventListeners(domainName, message);
+      }
```

Comment:
Why do we group the message orders by domain? If I understand correctly, for this list of incoming messages:

```
{domain: 'A', message: '1'}
{domain: 'B', message: '1'}
{domain: 'A', message: '2'}
{domain: 'B', message: '2'}
```

We'd "replay" them in this order:

```
{domain: 'A', message: '1'}
{domain: 'A', message: '2'}
{domain: 'B', message: '1'}
{domain: 'B', message: '2'}
```

Is this intentional?

Also, if one of the listener crashes, would we get missed messages stuck in the queue?